src/test/java/org/ays/auth/service/impl/AysUserUpdateServiceImplTest.java

```diff
@@ -128,6 +128,78 @@ void givenValidIdAndValidUpdateRequest_whenAllFieldsChanged_thenUpdateUser() {
                 .save(Mockito.any(AysUser.class));
     }
 
+    @Test
+    void givenValidIdAndValidUpdateRequest_whenRolesUpdatedOnly_thenUpdateUser() {
+
+        // Given
+        String mockId = "3c57d56b-4a97-4f70-86a9-b4c9235cbe13";
+
+        Set<String> mockRoleIds = Set.of(
+                "00a07704-8d7c-4048-b001-9fb69b22bfe8",
+                "7913e093-ae70-4029-ad8d-efbb21d79f26"
+        );
+        AysUserUpdateRequest mockUpdateRequest = new AysUserUpdateRequestBuilder()
+                .withValidValues()
+                .withRoleIds(mockRoleIds)
+                .build();
+
+        // When
+        Institution mockInstitution = new InstitutionBuilder()
+                .withValidValues()
+                .build();
+        Mockito.when(identity.getInstitutionId())
+                .thenReturn(mockInstitution.getId());
+
+        AysUser mockUser = new AysUserBuilder()
+                .withValidValues()
+                .withId(mockId)
+                .withFirstName(mockUpdateRequest.getFirstName())
+                .withLastName(mockUpdateRequest.getLastName())
+                .withEmailAddress(mockUpdateRequest.getEmailAddress())
+                .withPhoneNumber(AysPhoneNumber.builder().build())
+                .withCity(mockUpdateRequest.getCity())
+                .withRoles(List.of(
+                        new AysRoleBuilder()
+                                .withValidValues()
+                                .build()
+                ))
+                .withInstitution(mockInstitution)
+                .build();
+        Mockito.when(userReadPort.findById(Mockito.anyString()))
+                .thenReturn(Optional.of(mockUser));
+
+        List<AysRole> mockRoles = new ArrayList<>();
+        mockUpdateRequest.getRoleIds().forEach(roleId -> {
+            AysRole mockRole = new AysRoleBuilder()
+                    .withValidValues()
+                    .withId(roleId)
+                    .withInstitution(mockInstitution)
+                    .build();
+            mockRoles.add(mockRole);
+        });
+        Mockito.when(roleReadPort.findAllByIds(Mockito.anySet()))
+                .thenReturn(mockRoles);
+
+        Mockito.when(userSavePort.save(Mockito.any(AysUser.class)))
+                .thenReturn(Mockito.mock(AysUser.class));
+
+        // Then
+        userUpdateService.update(mockId, mockUpdateRequest);
+
+        // Verify
+        Mockito.verify(userReadPort, Mockito.times(1))
+                .findById(Mockito.anyString());
+
+        Mockito.verify(identity, Mockito.times(mockRoles.size() + 1))
```

Comment:
Burada her rol için gitmek yerine bir kere gitmek daha doğru olmaz mı? Değişmeyeceğini düşünüyorum.
```suggestion
        Mockito.verify(identity, Mockito.times(1))
```

Reply:
Optional.empty() durumu da olduğu için     bu kısmı ayırdım
  ```
final AysUser user = userReadPort.findById(id)
                .filter(userFromDatabase -> institutionId.equals(userFromDatabase.getInstitution().getId()))
                .orElseThrow(() -> new AysUserNotExistByIdException(id));
                
```